CPAC/nuisance/utils/compcor.py

```diff
@@ -92,17 +92,37 @@ def cosine_filter(
 ):
     """
     `cosine_filter` adapted from Nipype.
-
     https://github.com/nipy/nipype/blob/d353f0d/nipype/algorithms/confounds.py#L1086-L1107
-
-    Parameters
-    ----------
-    input_image_path : string
-            Bold image to be filtered.
+    
+    Parameters:
+    -----------
+    input_image_path : str
+        Path to the BOLD image to be filtered.
     timestep : float
-            'Repetition time (TR) of series (in sec) - derived from image header if unspecified'
-    period_cut : float
-            Minimum period (in sec) for DCT high-pass filter, nipype default value: 128.
+        Repetition time (TR) of the series (in seconds). Derived from image header if unspecified.
+    period_cut : float, optional
+        Minimum period (in seconds) for the DCT high-pass filter. Default value is 128.
+    remove_mean : bool, optional
+        Whether to remove the mean from the voxel time series before filtering. Default is True.
+    axis : int, optional
+        The axis along which to apply the filter. Default is -1 (last axis).
+    failure_mode : {'error', 'ignore'}, optional
+        Specifies how to handle failure modes. If set to 'error', the function raises an error.
+        If set to 'ignore', it returns the input data unchanged in case of failure. Default is 'error'.
+
+    Returns:
+    --------
+    cosfiltered_img : str
+        Path to the filtered BOLD image.
+
+    Notes:
+    ------
```

Comment:
Maybe I'm missing something, but why add colons to [the section headers](https://numpydoc.readthedocs.io/en/latest/format.html)?

```suggestion
    Returns
    -------
    cosfiltered_img : str
        Path to the filtered BOLD image.

    Notes
    -----
```